src/components/panel/panel.js

```diff
@@ -0,0 +1,365 @@
+ *
+ * Aligning the panel with the bottom of an element:
+ * `$mdPanelPosition.relativeTo(someElement);`
+ */
+
+/**
+ * @ngdoc method
+ * @name $mdPanelPosition#absolute
+ * @description
+ * Positions the panel absolutely relative to the parent element. If the parent is
+ * document.body, this is equivalent to positioning the panel absolutely within the
+ * viewport.
+ * @returns {$mdPanelPosition}
+ */
+
+/**
+ * @ngdoc method
+ * @name $mdPanelPosition#relativeTo
+ * @description
+ * Positions the panel relative to a specific element.
+ * @param {angular.JQLite} element Element to position the panel with respect to.
+ * @returns {$mdPanelPosition}
+ */
+
+/**
+ * @ngdoc method
+ * @name $mdPanelPosition#top
+ * @description
+ * Sets the value of `top` for the panel.
+ * @param {string} top Value of `top`.
+ * @returns {$mdPanelPosition}
+ */
+
+/**
+ * @ngdoc method
+ * @name $mdPanelPosition#bottom
+ * @description
+ * Sets the value of `bottom` for the panel.
+ * @param {string} top Value of `bottom`.
+ * @returns {$mdPanelPosition}
+ */
+
+/**
+ * @ngdoc method
+ * @name $mdPanelPosition#left
+ * @description
+ * Sets the value of `left` for the panel.
+ * @param {string} top Value of `left`.
+ * @returns {$mdPanelPosition}
+ */
+
+/**
+ * @ngdoc method
+ * @name $mdPanelPosition#right
+ * @description
+ * Sets the value of `right` for the panel.
+ * @param {string} top Value of `right`.
+ * @returns {$mdPanelPosition}
+ */
+
+/**
+ * @ngdoc method
+ * @name $mdPanelPosition#offsetX
+ * @description
+ * Sets the value of the offset in the x-direction.
+ * @param {string} offsetX
+ * @returns {$mdPanelPosition}
+ */
+
+/**
+ * @ngdoc method
+ * @name $mdPanelPosition#offsetY
+ * @description
+ * Sets the value of the offset in the y-direction.
+ * @param {string} offsetY
+ * @returns {$mdPanelPosition}
+ */
+
+
+/***************************************************************************************
+ *                                   IMPLEMENTATION                                    *
+ ***************************************************************************************/
+
+
+/**
+ * A service that is used for controlling/displaying panels on the screen.
+ * @param {!angular.JQLite} $rootElement
+ * @param {!angular.$injector} $injector
+ * @final @constructor @ngInject
+ */
+function MdPanelService($rootElement, $injector) {
+  // Default config options for the panel.
+  this.defaultConfigOptions_ = {
+    attachTo: $rootElement
+  };
+
+  this.config_ = this.defaultConfigOptions_;
+  this.$injector_ = $injector;
+}
+
+
+/**
+ * Creates a panel with the specified options.
+ * @param {!Object=} opt_config Configuration object for the panel.
+ * @returns {!$mdPanelRef}
+ */
+MdPanelService.prototype.create = function(opt_config) {
+  var configSettings = opt_config || {};
+  angular.forEach(configSettings, function(value, key) {
+    if (configSettings.hasOwnProperty(key)) {
```

Comment:
`angular.forEach` does it's own filtering on `hasOwnProperty`, so you don't have to do it here.